libraries/AP_HAL_PX4/CAN.cpp

```diff
@@ -256,31 +256,43 @@ int PX4CAN::computeTimings(const uint32_t target_bitrate, Timings& out_timings)
     // First attempt with rounding to nearest
     BsPair solution(bs1_bs2_sum, uint8_t(((7 * bs1_bs2_sum - 1) + 4) / 8));
 
-    if (solution.sample_point_permill > MaxSamplePointLocation) {
-        // Second attempt with rounding to zero
-        solution = BsPair(bs1_bs2_sum, uint8_t((7 * bs1_bs2_sum - 1) / 8));
-    }
+    if (solution.isValid())
+    {
+        if (solution.sample_point_permill > MaxSamplePointLocation) {
```

Comment:
That's was my point initially. The solution.sample_point_permill > MaxSamplePointLocation is the only check we actually need.
But I see your point and will revert to proposed way.